solutions/swb-reference/integration-tests/tests/multiStep/environment.test.ts

```diff
@@ -29,6 +36,196 @@ describe('multiStep environment test', () => {
+
+    //Verify Completed Environment B
+    console.log('Verify Completed Environment B');
+    await adminSession.resources.environments
+      .environment(environmentB.id)
+      .pollEnvironment(15, ENVIRONMENT_START_MAX_WAITING_SECONDS, (env) => env.status === 'PENDING'); //wait for environmentB to complete
+    const { data: environmentBCompleted } = await adminSession.resources.environments
+      .environment(environmentB.id)
+      .get();
+    expect(environmentBCompleted).toMatchObject({
+      id: expect.stringMatching(uuidRegExp),
+      status: 'COMPLETED',
+      ETC: expect.anything(), //ETC should be defined
+      PROJ: expect.anything() // PROJ should be defined
+    });
+    console.log('Environment B Completed');
+
+    //Verify Connect Environment A
+    console.log('Verify Connect Environment A');
+    const { data: environmentAConnectInfo } = await adminSession.resources.environments
+      .environment(environmentA.id)
+      .connect();
+    expect(environmentAConnectInfo).toMatchObject({
+      authCredResponse: {
+        url: expect.anything()
+      },
+      instructionResponse: expect.anything()
+    });
+
+    //Stop Environment A
+    console.log('Stopping Environment A');
+    await adminSession.resources.environments.environment(environmentA.id).stop();
+    await adminSession.resources.environments
+      .environment(environmentA.id)
+      .pollEnvironment(15, ENVIRONMENT_STOP_MAX_WAITING_SECONDS, (env) => env.status === 'STOPPING'); //wait for environmentA to stop
+    const { data: environmentAStopped } = await adminSession.resources.environments
+      .environment(environmentA.id)
+      .get();
+    expect(environmentAStopped).toMatchObject({
+      id: expect.stringMatching(uuidRegExp),
+      status: 'STOPPED',
+      ETC: expect.anything(), //ETC should be defined
+      PROJ: expect.anything() // PROJ should be defined
+    });
+    console.log('Environment A Stopped');
+
+    //Search Environment A filtering by name
+    console.log('Searching Environment A filtering by name');
+    const { data: environmentsNameFilter } = await adminSession.resources.environments.get({
+      name: environmentAStopped.name
+    });
+    expect(
+      //eslint-disable-next-line @typescript-eslint/no-explicit-any
+      environmentsNameFilter?.data?.filter((env: any) => env.id === environmentAStopped.id)?.length
+    ).toBeTruthy();
+
+    //Search Environment A filtering by status
+    console.log('Searching Environment A filtering by status');
+    const { data: environmentsStatusFilter } = await adminSession.resources.environments.get({
+      status: environmentAStopped.status
+    });
+    expect(
+      //eslint-disable-next-line @typescript-eslint/no-explicit-any
+      environmentsStatusFilter?.data?.filter((env: any) => env.id === environmentAStopped.id)?.length
+    ).toBeTruthy();
+
+    //Search Environment A filtering by created at
+    console.log('Searching Environment A filtering by created at');
```

Comment:
`Searching for Environment A: filtering by "createdAt"`

Same format for the other `console.log`